tests/foreman/ui/test_ansible.py

```diff
@@ -177,3 +180,84 @@ def test_positive_config_report_ansible(session, target_sat, module_org, rhel_co
         assert len(session.configreport.read()['table']) == 0
+
+
+@pytest.mark.no_containers
+@pytest.mark.rhel_ver_match('9')
+def test_positive_ansible_custom_role(target_sat, session, module_org, rhel_contenthost):
+    """
+    Test Config report generation with Custom Ansible Role
+
+    :id: 3551068a-ccfc-481c-b7ec-8fe2b8a802bf
+
+    :customerscenario: true
+
+    :Steps:
+
+        1. Register a content host with satellite
+        2. Create a  custom role and import  into satellite
+        3. Assign that role to a host
+        4. Assert that the role was assigned to the host successfully
+        5. Run the Ansible playbook associated with that role
+        6. Check if the report is created successfully
+
+    :expectedresults:
+        1. Config report should be visible for a custom role run.
+
+    :BZ: 2155392
+
+    :CaseAutomation: Automated
+    """
+    SELECTED_ROLE = 'custom_role'
+    _, test_file = mkstemp(prefix='task-', suffix='.yaml')
+    data = {
+        "name": "Copy ssh keys",
+        "copy": {
+            "src": "/var/lib/foreman-proxy/ssh/{{ item }}",
+            "dest": "/root/.ssh",
+            "owner": "root",
+            "group": "root",
+            "mode": "0400",
+        },
+        "with_items": ["id_rsa_foreman_proxy.pub", "id_rsa_foreman_proxy"],
+    }
+    with open(test_file, "w") as f:
+        yaml.dump(data, f, sort_keys=False, default_flow_style=False)
+    target_sat.execute('mkdir /etc/ansible/roles/custom_role')
+    target_sat.put(test_file, '/etc/ansible/roles/custom_role/test.yaml')
+    rhel_contenthost.install_katello_ca(target_sat)
+    rhel_contenthost.register_contenthost(module_org.label, force=True)
+    assert rhel_contenthost.subscribed
+    rhel_contenthost.add_rex_key(satellite=target_sat)
+    proxy_id = target_sat.nailgun_smart_proxy.id
+    target_host = rhel_contenthost.nailgun_host
+    target_sat.api.AnsibleRoles().sync(data={'proxy_id': proxy_id, 'role_names': [SELECTED_ROLE]})
+    target_host.assign_ansible_roles(data={'ansible_role_ids': [1]})
+    host_roles = target_host.list_ansible_roles()
+    assert host_roles[0]['name'] == SELECTED_ROLE
+
+    template_id = (
+        target_sat.api.JobTemplate()
+        .search(query={'search': 'name="Ansible Roles - Ansible Default"'})[0]
+        .id
+    )
+    job = target_sat.api.JobInvocation().run(
+        synchronous=False,
+        data={
+            'job_template_id': template_id,
+            'targeting_type': 'static_query',
+            'search_query': f'name = {rhel_contenthost.hostname}',
+        },
+    )
+    target_sat.wait_for_tasks(
+        f'resource_type = JobInvocation and resource_id = {job["id"]}', poll_timeout=1000
+    )
+    result = target_sat.api.JobInvocation(id=job['id']).read()
+    assert result.succeeded == 1
+    with session:
+        session.organization.select(module_org.name)
+        session.location.select(constants.DEFAULT_LOC)
```

Comment:
The org should already be set to module_org when the test session starts. If you want to use module_location then you won't have to select default location either. See https://github.com/SatelliteQE/robottelo/blob/master/pytest_fixtures/core/ui.py#L23

Reply:
I can try removing the org here, about the location I think we should have the default one here as with the core/ui I don't want the module_loc to be selected by default.